### ISSUE TITLE
nREPL version available in lein template is 0.2.10

### ISSUE THREAD

**Issue by novakboskov** [technomancy/leiningen#2072]
When make a leiningen template with ```lein new template my-template``` and then launch repl it say that available nREPL version is 0.2.10. In project.clj of Leiningen itself nREPL version 0.2.12 is listed as a dependency. Templates have ```:eval-in-leiningen true```, why don't ```my-template``` use nREPL 0.2.12 over 0.2.10?

```lein version``` is ```Leiningen 2.5.3 on Java 1.8.0_66 Java HotSpot(TM) 64-Bit Server VM```
~/.lein/profiles.clj is:
```
{:user
 {:plugins [[environ/environ.lein "0.3.0"]
            [lein-set-version "0.4.1"]
            [lein-simpleton "1.1.0"]
            [lein-try "0.4.1"]
            [lein-ancient "0.6.8-SNAPSHOT"]
            [cider/cider-nrepl "0.11.0-SNAPSHOT"]
            [refactor-nrepl "2.0.0-SNAPSHOT"]]
  :dependencies [[org.clojure/tools.nrepl "0.2.12"]]

  :hooks [environ.leiningen.hooks]}}

```

**Comment by hyPiRion** [comment]
The nREPL version has been bumped to 0.2.12  after 2.5.3 came out. 2.5.3 ships with 0.2.10. When 2.6.0 is out, the default version will be 0.2.12.

If you want to upgrade to `0.2.12` by default, see [Replacing Default Repl Dependencies](https://github.com/technomancy/leiningen/blob/master/doc/PROFILES.md#replacing-default-repl-dependencies). In your case, it should be sufficient to do

```clj
{:user
 {:plugins [[environ/environ.lein "0.3.0"]
            [lein-set-version "0.4.1"]
            [lein-simpleton "1.1.0"]
            [lein-try "0.4.1"]
            [lein-ancient "0.6.8-SNAPSHOT"]
            [cider/cider-nrepl "0.11.0-SNAPSHOT"]
            [refactor-nrepl "2.0.0-SNAPSHOT"]]

  :hooks [environ.leiningen.hooks]}
 :repl {:dependencies [[org.clojure/tools.nrepl "0.2.12"]]}}
```

**Comment by novakboskov** [comment]
Hm... It dosn't work. ```lein repl``` in template directory still reports:
```
nREPL server started on port 43705 on host 127.0.0.1 - nrepl://127.0.0.1:43705
REPL-y 0.3.7, nREPL 0.2.10
...
```
Version 0.2.12 is respected only in projects without ```:eval-in-leiningen true``` and outside any project.
I have also tried to put
```
  :repl {:dependencies [[org.clojure/tools.nrepl "0.2.12"]]}
```
in template's project.clj and it still use 0.2.10.

**Comment by hyPiRion** [comment]
Ah, right, this is happening with `:eval-in-leiningen`.

I don't think there is any straightforward way to do this, as Leiningen itself embeds a version of `tools.nrepl`.

**Comment by hyPiRion** [comment]
Either way, this will be fixed once 2.6.0 is out.

**Comment by hyPiRion** [comment]
Fixed through release of 2.6.0.
